### ISSUE TITLE
confusion matrix plotting code maybe wrong

### ISSUE THREAD

**Issue by lichao977** [deepcam-cn/yolov5-face#296]
I found that the plotting code of the confusion matrix is something wrong. 

https://github.com/deepcam-cn/yolov5-face/blob/152c688d551aefb973b7b589fb0691c93dab3564/test.py#L135
            if len(pred) == 0:
                if nl:
                    stats.append((torch.zeros(0, niou, dtype=torch.bool), torch.Tensor(), torch.Tensor(), tcls))
                continue
            
When the detection result  of an image is None,   confusion_matrix.process_batch is not added to deal with it.  So the confusion matrix will not take account into the empty detection images. 

I have checked the related codes in yolov5, they actually had the process_batch manipulation. 
            if npr == 0:
                if nl:
                    stats.append((correct, *torch.zeros((2, 0), device=device), labels[:, 0]))
                    if plots:
                        confusion_matrix.process_batch(detections=None, labels=labels[:, 0])
                continue
